### PR TITLE
Add Custom Properties Support for Authentication Portal

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
### Purpose
Adding logic to copy custom properties to web.xml file from deployment.toml

### Sample Usage
in deployment.toml
```
[authenticationendpoint.context_params]
sample_prop1=value1
sample_prop2=value2

```

In web.xml (after loading params)
```
<context-param>
    <param-name>sample_prop1</param-name>
    <param-value>"value1"</param-value>
</context-param>

<context-param>
    <param-name>sample_prop2</param-name>
    <param-value>"value2"</param-value>
</context-param>
```

### Checklist
- [x] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by brionmario — APPROVED** [review]

File: features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2 [hunk 1]
```diff
@@ -125,6 +125,17 @@
         <param-value>true</param-value>
     </context-param>
 
+    <!-- *************** Authentication Portal Custom Configurations ********************** -->
+    {% if authenticationendpoint_custom_properties is defined %}
+    {% for key, value in authenticationendpoint_custom_properties.items() %}
```
Comment (sumedhe) [inline features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:130]: `authenticationendpoint_custom_properties` -> `authenticationendpoint_context_params` WDYT?

Comment (ThaminduDilshan) [reply features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2]: fixed. Config changed to `authenticationendpoint.context_params` as discussed

**Review by sumedhe — APPROVED** [review]

File: features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2 [hunk 2]
```diff
@@ -125,6 +125,17 @@
         <param-value>true</param-value>
     </context-param>
 
+    <!-- *************** Authentication Portal Context Parameters ********************** -->
+    {% if authenticationendpoint.context_params is defined %}
+    {% for key, value in authenticationendpoint.context_params.items() %}
+    <context-param>
+        <param-name>{{key}}</param-name>
+        <param-value>"{{value}}"</param-value>
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:133]: Yes will need

[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:133]: Can remove with a change in the places where this parameter is being used.
Fixed in #2733

**Pull request merged**